rust/src/delta_datafusion.rs

```diff
@@ -373,25 +375,44 @@ impl TableProvider for DeltaTable {
             });
         };
 
-        let table_partition_cols = self.get_metadata()?.partition_columns.clone();
+        let table_partition_cols = self
+            .get_metadata()?
+            .partition_columns
+            .clone()
+            .into_iter()
+            .flat_map(|c| {
+                let f_types = schema
+                    .fields()
+                    .iter()
+                    .map(|e| (e.name(), e.data_type()))
+                    .filter(|&(col_name, _)| c.eq_ignore_ascii_case(col_name))
+                    .collect::<Vec<_>>();
```

Comment:
maybe use `schema.field_with_name(col)?.data_type()` here to avoid the iteration?